task/build-vm-image/0.1/build-vm-image.yaml

```diff
@@ -148,6 +154,24 @@ spec:
         echo "$BUILD_DIR"
         ssh -v $SSH_ARGS "$SSH_HOST"  mkdir -p "$BUILD_DIR/workspaces" "$BUILD_DIR/scripts" "$BUILD_DIR/tmp" "$BUILD_DIR/tekton-results" "$BUILD_DIR/entitlement"
 
+        if [! -n "${CONFIG_TOML_FILE}" ]
+          echo "No CONFIG_TOML_FILE specified"
+          if [ -f /var/workdir/source/config.toml ]; then
+            echo "Using the config.toml file found in the repository root!"
+            echo "  Remove the config.toml file or set params.CONFIG_TOML_FILE to an invalid path to prevent its use."
+          else
+            echo "No config.toml file found. Set params.CONFIG_TOML_FILE to use one in the repository."
+            export CONFIG_TOML_FILE=config.toml
+          fi
+        fi
+        # ensure that a config toml file is present in case one is not provided or the path is invalid
+        mkdir -p /var/workdir/source/$(dirname $CONFIG_TOML_FILE)
+        touch /var/workdir/source/$CONFIG_TOML_FILE
```

Comment:
> or the path is invalid

The point of the changes suggested in https://github.com/konflux-ci/build-definitions/pull/1171#discussion_r1682771804 was to not do this (if the user specifies a wrong path, the task should fail). Otherwise the current code is just a more complex way to achieve the same behavior as before :smile: